docs/01-app/01-getting-started/07-server-and-client-components.mdx

```diff
@@ -0,0 +1,494 @@
+
+This page will go through how Server and Client Components work, and common examples of when to use them.
+
+## How do Server and Client Components work?
+
+### On the server
+
+On the server, Next.js uses React's APIs to orchestrate rendering. The rendering work is split into chunks, by individual route segments ([layouts and pages](/docs/app/getting-started/layouts-and-pages)):
+
+- **Server Components** are rendered into a special data format called the React Server Component Payload (RSC Payload).
+- **Client Components** are rendered into HTML.
+
+> **What is the React Server Component Payload (RSC)?**
+>
+> The RSC Payload is a compact binary representation of the rendered React Server Components tree. It's used by React on the client to update the browser's DOM. The RSC Payload contains:
+>
+> - The rendered result of Server Components
+> - Placeholders for where Client Components should be rendered and references to their JavaScript files
+> - Any props passed from a Server Component to a Client Component
+
+### On the client (first load)
+
+Then, on the client:
+
+1. **HTML** is used to immediately show a fast non-interactive preview of the route to the user.
+2. **RSC Payload** is used to reconcile the Client and Server Component trees.
+3. **JavaScript** is used to hydrate Client Components and make the application interactive.
+
+> **What is hydration?**
+>
+> Hydration is React's process for attaching [event handlers](https://react.dev/learn/responding-to-events) to the DOM, to make the static HTML interactive.
+
+### Subsequent Navigations
+
+On subsequent navigations:
+
+- The **RSC Payload** is prefetched and cached for instant navigation.
+- **Client Components** are rendered entirely on the client, without the server-rendered HTML.
+
+## When to use Server and Client Components?
+
+Use **Client Components** when you need:
+
+- [State](https://react.dev/learn/managing-state) and [event handlers](https://react.dev/learn/responding-to-events). E.g. `onClick`, `onChange`.
+- [Lifecycle logic](https://react.dev/learn/lifecycle-of-reactive-effects). E.g. `useEffect`.
+- Browser-only APIs. E.g. `localStorage`, `window`, `Navigator.geolocation`, etc.
+- [Custom hooks](https://react.dev/learn/reusing-logic-with-custom-hooks).
+
+Use **Server Components** when you need:
+
+- Fetch data from databases or APIs close to the source.
+- Prevent API keys, tokens, and other secrets from being exposed to the client.
+- Reduce the amount of JavaScript sent to the browser.
+- Improve the [First Contentful Paint (FCP)](https://web.dev/fcp/), and stream content progressively to the client.
+- Improve SEO by rendering content on the server.
+
```

Comment:
Before you get into the next topic, I think it's worth dedicating a paragraph to stating that this isn't an "either or" situation and you can mix and match them in a single tree without losing any features.

Reply:
Update the wording in the page intro and start of this section to capture more of this sentiment.